tests/CMakeLists.txt

```diff
@@ -238,6 +238,41 @@ if(Boost_FOUND)
   endif()
 endif()
 
+# Check if we need to add -lstdc++fs or -lc++fs or nothing
+if(MSVC)
```

Comment:
There's no need to special-case MSVC if you specify the standard version the way `try_compile` specifies in the docs: https://cmake.org/cmake/help/latest/command/try_compile.html

Reply:
thanks.

Reply:
I think this is fine as-is for the purposes of this PR; if an MSVC user encounters an issue, I'm sure we can handle it more acutely.